app/src/main/java/com/android/ubclaunchpad/driver/session/SessionInfoActivity.java

```diff
@@ -41,14 +48,31 @@ protected void onCreate(Bundle savedInstanceState) {
         final String sessionName = getIntent().getStringExtra(StringUtils.SESSION_NAME);
         TextView textViewSessionName = (TextView) findViewById(R.id.viewSessionName);
         textViewSessionName.setText(sessionName);
-
-        final DatabaseReference session = FirebaseUtils.getDatabase()
+        final Button goButton = (Button) findViewById(R.id.go_Button);
+        goButton.setVisibility(View.INVISIBLE);
+        session = FirebaseUtils.getDatabase()
                 .child(StringUtils.FirebaseSessionEndpoint)
                 .child(sessionName);
         final String UID = FirebaseUtils.getFirebaseUser().getUid();
         final User currentUser;
         try {
             currentUser = UserManager.getInstance().getUser();
+
+            session.addListenerForSingleValueEvent(new ValueEventListener() {
+                @Override
+                public void onDataChange(DataSnapshot dataSnapshot) {
+                    String uid = dataSnapshot.child("sessionHostUid").getValue(String.class);
+                    if (uid.equals(FirebaseUtils.getFirebaseUser().getUid())) {
```

Comment:
This is causing the app to crash for me when I try to join some of the sessions and the error message is that I'm calling equals on a null object. I'm guessing it's because dataSnapshot.child("sessionHostUid").getValue(String.class) is returning null because we just added the sessionHostUid endpoint and the older sessions don't actually have it. No big deal, I'd suggest adding the endpoint manually to the older sessions.